### PR TITLE
[aes/cov] Fix AES branch coverage by adding exclusions

### PR DESCRIPTION
**Pull request by abdullahvarici**
Exclude DV related code and a few parameter controlled branches.

Branch coverage for unmasked AES is increased from 83% to 97%.

Signed-off-by: Abdullah Varici <abdullah.varici@lowrisc.org>

### CODE REVIEW

**Comment by vogelpi** [comment]
Thanks for the PR @abdullahvarici , I've removed @matutem as reviewer for the moment. We should look at this together first. After having had a quick first look I have the impression that the new exclusion file excludes too many things.

**Comment by vogelpi** [comment]
Regarding the coverage exclusion file, @andreaskurth left some very useful comments in #16432 that also apply to this PR. Can you please ensure to update this PR accordingly?

**Comment by abdullahvarici** [comment]
Thanks @vogelpi. I updated the PR.

**Comment by abdullahvarici** [comment]
I've updated the PR to use masked unr file for unmasked design too. And exclude only masking related parts in unmasked unr file. Can you review it @vogelpi?

File: hw/ip/aes/dv/aes_masking_sim_cfg.hjson [hunk 1]
```diff
@@ -11,7 +11,7 @@
   import_cfgs: ["{proj_root}/hw/ip/aes/dv/aes_base_sim_cfg.hjson"]
 
   // Coverage exclusion
-  xcelium_cov_refine_files: ["{proj_root}/hw/ip/aes/dv/cov/aes_UNR.vRefine"]
+  xcelium_cov_refine_files: ["{proj_root}/hw/ip/aes/dv/cov/refines/aes_UNR.vRefine"]
```
Comment (vogelpi) [inline hw/ip/aes/dv/aes_masking_sim_cfg.hjson:14]: Thanks for catching this! I just checked that this also increases coverage numbers for the masked implementation as expected:
|  Score  |  Block  |  Branch  |  Statement  |  Expression  |  Toggle  |   Fsm   |  Assertion  |  CoverGroup  |
|:-------:|:-------:|:--------:|:-----------:|:------------:|:--------:|:-------:|:-----------:|:------------:|
| 96.10  | 95.83 | 90.86 | 98.35     | 92.98      | 96.25  | 97.78 | 98.67     | 92.87      | 
| 96.76 | 96.52 | 91.98  |   98.62   |   94.66    | 96.33  | 97.90 |   98.39   |   91.85    |

**Review by vogelpi — APPROVED** [review]
Thanks @abdullahvarici for working on this! The changes look good, I've reviewed the exclusions individually (thanks to your chosen approach this was feasible) and they all make sense. LGTM!

**Comment by vogelpi** [comment]
I just got the coverage results with that PR also for the unmasked implementation:
|  Score  |  Block  |  Branch  |  Statement  |  Expression  |  Toggle  |   Fsm   |  Assertion  |  CoverGroup  |
|:-------:|:-------:|:--------:|:-----------:|:------------:|:--------:|:-------:|:-----------:|:------------:|
| 93.08 | 91.81 | 84.18 | 96.37 | 90.12 | 99.03 | 90.37 | 98.26 | 91.45
| 94.80 | 94.04 | 87.58 | 97.53 |   91.72    | 99.14  | 91.61 |   97.71   |   91.85    |

It improves notably, but we're not yet fully there.

**Comment by vogelpi** [comment]
We should probably also add exclusions for the INV operation of the S-Boxes inside the key expand module. In addition, I noticed some warnings when loading the masked coverage exclusion file, probably due to RTL changes I did during the CSRNG V2S push. But we can fix this as a follow-up. I am merging this now.

**Pull request merged**